### PR TITLE
Configure Apache as an s6 service

### PR DESCRIPTION
**Pull request by sameersbn**
Changes: 
 - Apache is configured as an S6 service
 - Do not start apache if user specifies a command at launch
 - Allow users to pass arguments to httpd
 - Added circuit-breaker logic to catch repeated failures of apache service
 - Added `--noproxy apache` to `curl` in `tests.sh`

### CODE REVIEW

File: rootfs/etc/services.d/apache/finish [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/with-contenv bash
```
Comment (prydonius) [inline rootfs/etc/services.d/apache/finish]: we may want to have this in the base image and just copy it to the right directory

**Comment by prydonius** [comment]
add to whitelist

File: rootfs/etc/services.d/apache/finish (hunk 1 above)
Comment (sameersbn) [inline rootfs/etc/services.d/apache/finish]: you mean the finish script?

Comment (prydonius) [inline rootfs/etc/services.d/apache/finish]: yep I meant the whole script

That would work too! I'll merge this for now though

**Pull request merged**